vendor/k8s.io/kubernetes/pkg/controller/daemon/metrics.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package daemon
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	daemonControllerSubsystem = "daemon_controller"
+	podFailedPerNodeNumberKey = "pods_failed_per_node_number"
+)
+
+var (
+	podFailedPerNodeNumber = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: daemonControllerSubsystem,
+			Name:      podFailedPerNodeNumberKey,
+			Help:      "Number of pods observed as failed for particular daemon set per node.",
+		},
+		[]string{"daemon", "node_name"},
```

Comment:
I did not expect a metric with a label per resource... the cardinality is way too high, right?

Reply:
basically the same comment as https://github.com/kubernetes/kubernetes/pull/65215#issuecomment-398406184

Reply:
@liggitt @smarterclayton was ok with the cardinality